src/main/java/org/fhirvp/notification/impl/NotificationHandlerImpl.java

```diff
@@ -0,0 +1,31 @@
+package org.fhirvp.notification.impl;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.quarkus.arc.Unremovable;
+import org.fhirvp.notification.FHIRNotificationEvent;
+import org.fhirvp.notification.NotificationHandler;
+
+import javax.enterprise.context.ApplicationScoped;
+
+@ApplicationScoped
+@Unremovable
+public class NotificationHandlerImpl implements NotificationHandler {
+
+    @Override
+    public void handleNotification(String notificationJson) {
+        ObjectMapper objectMapper = new ObjectMapper();
+        try {
+            handleNotification(objectMapper.readValue(notificationJson, FHIRNotificationEvent.class));
+        } catch (JsonProcessingException e) {
+            e.printStackTrace();
+        }
+    }
```

Comment:
I think the `NotificationHandler` should not offer `handleNotification(String notificationJson)`. It's the responsibility of the API/WebSocket layer to convert data that arrives to the domain objects. This also allows us to have better error handling than just printing the stack trace of the exception, for instance in the API layer we can return `HTTP 400: Bad Request` if the JSON body sent to the endpoint does not follow the `FHIRNotificationEvent` structure.

Reply:
Addressed at 54ec843